src/Model/Behavior/SearchBehavior.php

```diff
@@ -101,7 +103,14 @@ public function searchManager()
      */
     protected function _extractParams($params, $filters)
     {
-        return array_intersect_key(Hash::filter($params), $filters);
+        return array_intersect_key(Hash::filter($params, function ($val) {
+            foreach ($this->getConfig('emptyValues') as $emptyValue) {
+                if ($val === $emptyValue) {
```

Comment:
What about null? Was that filtered out so far, then emptyValues default above needs to also contain that one for BC.

Reply:
For BC, I can add following to the default config:
```
'emptyValues' => ['',null]
```

What else should I add?

Reply:
Before: https://api.cakephp.org/3.0/source-class-Cake.Utility.Hash.html#618 (3.0)
https://api.cakephp.org/3.5/source-class-Cake.Utility.Hash.html#643-652 (3.5)

Aka: if `($var === 0 || $var === '0' || !empty($var))` (3.0) and `0.0` (3.5) https://api.cakephp.org/3.5/source-class-Cake.Utility.Hash.html#643-652

For empty() see: https://3v4l.org/sj8bM

So all these "generally" need to be filtered by default:

```php
    '',
    null,
    false,
    0,
    0.0,
    '0',
```

Minus the exception of `Hash::filter()`'s defaults: `$var === 0 || $var === '0'`
And since some time `0.0` https://api.cakephp.org/3.5/source-class-Cake.Utility.Hash.html#643-652

Leaves these as a requirement to be filtered out by `'emptyValues'` by convention/default:

```php
    '',
    null,
    false,
```

p.s.: We should also have an option to ltrim/rtrim inputs if it is not there already.

Reply:
Just make sure it behaves the same by default as the current Hash::filter() to avoid BC issues.

Reply:
Not sure why my comment appears twice.